File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> st;
        int left=0,right=0,maxLen=0;
        while(right<s.size())
        {
            if(!st.count(s[right])) {
                st.insert(s[right]);
                maxLen = max(maxLen,right-left+1);
                right++;
            } else{
                st.erase(s[left]);
                left++;
            }
        }
        return maxLen;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int left = 0;
        int right = 0;
        int maxLen = Math.min(s.length(),1);
        HashSet<Character> set  = new HashSet<>();
        for(right=0;right<s.length();right++)
        {
           char ch = s.charAt(right);
           while(set.contains(ch))
           {
            set.remove(s.charAt(left));
            left++;
           }
           set.add(ch);
           int window = right-left+1;
           maxLen=Math.max(window,maxLen);
        }
        return maxLen;
    }
}